plugins/inputs/cloudwatch/cloudwatch.go

```diff
@@ -410,27 +410,29 @@ func (c *CloudWatch) fetchNamespaceMetrics() ([]*cwClient.Metric, error) {
 	default:
 		recentlyActive = nil
 	}
-	params = &cwClient.ListMetricsInput{
-		Dimensions:     []*cwClient.DimensionFilter{},
-		NextToken:      token,
-		MetricName:     nil,
-		RecentlyActive: recentlyActive,
-	}
 	for _, namespace := range c.Namespaces {
-		params.Namespace = aws.String(namespace)
+
+		params = &cwClient.ListMetricsInput{
+			Dimensions:     []*cwClient.DimensionFilter{},
+			NextToken:      token,
+			MetricName:     nil,
+			RecentlyActive: recentlyActive,
+	    	}
+	    	params.Namespace = aws.String(namespace)
```

Comment:
This can be simplified to something like
```
params = &cwClient.ListMetricsInput {
    Namespace:      aws.String(namespace),
    Dimensions:     []*cwClient.DimensionFilter{},
    NextToken:      token,
    MetricName:     nil,
    RecentlyActive: recentlyActive,
}
```